shared/lib/themes/actions.js

```diff
@@ -1,99 +1,90 @@
 /**
  * External dependencies
  */
-var page = require( 'page' ),
-	defer = require( 'lodash/function/defer' ),
-	debug = require( 'debug' )( 'calypso:themes:actions' ); //eslint-disable-line no-unused-vars
+import page from 'page';
+import defer from 'lodash/function/defer';
+import debugFactory from 'debug';
+const debug = debugFactory( 'calypso:themes:actions' ); //eslint-disable-line no-unused-vars
 
 /**
  * Internal dependencies
  */
-var Dispatcher = require( 'dispatcher' ),
-	wpcom = require( 'lib/wp' ),
-	analytics = require( 'analytics' ),
-	ThemesStore = require( './stores/themes' ),
-	ThemesListStore = require( './stores/themes-list' ),
-	ThemeConstants = require( './constants' ),
-	ThemeHelpers = require( './helpers' ),
-	CurrentThemeStore = require( './stores/current-theme' ),
-	LastEventStore = require( './stores/themes-last-event' ),
-	ThemeConstants = require( './constants' ),
-	ThemesLastQueryStore = require( './stores/themes-last-query' );
-
-var ThemesActions = {
-	fetch: function( site ) {
-		const queryParams = ThemesListStore.getQueryParams();
+import ThemeConstants from 'lib/themes/constants';
+import ThemeHelpers from './helpers';
+import { getCurrentTheme } from './reducers/current-theme';
+import { getThemeById } from './reducers/themes';
+import { getQueryParams } from './reducers/themes-list';
+import { hasSiteChanged, hasParams } from './reducers/themes-last-query';
+import wpcom from 'lib/wp';
+
+export function fetchThemes( site ) {
+	return ( dispatch, getState ) => {
+		const queryParams = getQueryParams( getState().themes.themesList );
 		const startTime = new Date().getTime();
-		const callback = function( error, data ) {
+		const callback = ( error, data ) => {
 			debug( 'Received themes data', data );
 			if ( error ) {
-				this.receiveServerError( error );
+				dispatch( receiveServerError( error ) );
 			} else {
 				const responseTime = ( new Date().getTime() ) - startTime;
-				this.receiveThemes( data, site, queryParams, responseTime );
+				dispatch( receiveThemes( data, site, queryParams, responseTime ) );
 			}
-		}.bind( this );
+		};
 
 		debug( 'Query params', queryParams );
 		wpcom.undocumented().themes( site, queryParams, callback );
-	},
+	}
+}
 
-	fetchJetpack: function( site ) {
-		if ( hasSiteChanged() ) {
-			return this.fetch( site );
+/**
+ * On Jetpack sites, we're not querying themes using the REST API, but fetch all of the installed
+ * themes (which typically aren't that many), and filter them within Calypso.
```

Comment:
> (which typically aren't that many)

We can probably lose the part of the comment in brackets without sacrificing any information.